llvm/lib/Transforms/Scalar/LoopIdiomRecognize.cpp

```diff
@@ -1473,6 +1491,160 @@ bool LoopIdiomRecognize::avoidLIRForMultiBlockLoop(bool IsMemset,
   return false;
 }
 
+bool LoopIdiomRecognize::optimizeCRCLoop(const PolynomialInfo &Info) {
+  // FIXME: Hexagon has a special HexagonLoopIdiom that optimizes CRC using
```

Comment:
We should add a new TTI hook instead of checking the target.

Reply:
I think there is already work in progress to add a TTI hook with the CLMUL intrinsic work (https://github.com/llvm/llvm-project/pull/140301), and @nikic suggested this solution as a stopgap.